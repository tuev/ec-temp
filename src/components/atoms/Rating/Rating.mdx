import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './Rating.stories.tsx'
import Rating from './Rating.tsx'

<Story name="basic" />
