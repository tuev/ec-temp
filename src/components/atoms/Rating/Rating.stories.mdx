import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './Rating.stories.tsx'
import Rating from './Rating.tsx'

<Meta title="Rating" component={Rating} />
