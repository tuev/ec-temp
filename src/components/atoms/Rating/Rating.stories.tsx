import React from 'react'
import { action } from '@storybook/addon-actions'
import Rating from './Rating'

export default {
  component: Rating,
  title: 'AppRating',
}

export const basic = () => (
  <>
    <Rating onChange={action('Rating Clicked!!!')} value={4.3} />
  </>
)


