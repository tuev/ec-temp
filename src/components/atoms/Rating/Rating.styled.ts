import { space } from 'styled-system'
import styled from 'styled-components'

import { Rating } from '@material-ui/lab'
import { RatingDefaultProps } from './Rating.types'

export const AppRating: React.FC<RatingDefaultProps> = styled(Rating)<
  RatingDefaultProps
>`
  ${space}
`
