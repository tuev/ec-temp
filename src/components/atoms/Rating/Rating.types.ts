import { SpaceProps } from 'styled-system'
import { RatingProps } from '@material-ui/lab/Rating'

export type HandleRatingChange = (value: number) => unknown

export type OnRatingChange = (
  e: React.ChangeEvent<{}>,
  value: number | null
) => void

export type RatingHook = (
  value: number,
  onValueChange?: HandleRatingChange
) => [number, OnRatingChange]

export interface IRatingAppProps {
  value: number
}

export type RatingDefaultProps = RatingProps & SpaceProps

export declare type BaseRatingProps = Omit<
  RatingDefaultProps,
  'value' | 'onChange'
> &
  IRatingAppProps & { onChange?: HandleRatingChange }
