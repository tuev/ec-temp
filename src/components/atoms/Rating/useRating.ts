import { useState, useCallback, useEffect } from 'react'
import { RatingHook } from './Rating.types'

const useRating: RatingHook = (rating, onRatingChange) => {
  const [value, onValueChange] = useState<number>(rating)

  const handleValueChange = useCallback(
    (event: React.ChangeEvent<{}>, newValue: number | null): void => {
      const valueRating = newValue || value
      onValueChange(valueRating)
    },
    [value]
  )

  useEffect(() => {
    if (onRatingChange) {
      onRatingChange(value)
    }
  }, [onRatingChange, value])

  return [value, handleValueChange]
}

export default useRating
