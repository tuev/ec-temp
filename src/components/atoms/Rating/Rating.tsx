import React, { FC } from 'react'
import { AppRating } from './Rating.styled'
import { BaseRatingProps } from './Rating.types'
import useRating from './useRating'

const Rating: FC<BaseRatingProps> = (props: BaseRatingProps) => {
  const { value, onChange } = props
  const [rate, onRateChange] = useRating(value, onChange)
  return (
    <AppRating
      {...props}
      value={rate}
      onChange={onRateChange}
      name="app-rating"
    />
  )
}

export default Rating
