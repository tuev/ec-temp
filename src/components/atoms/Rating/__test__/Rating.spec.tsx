import React from 'react'
import Rating from '..'
import { mount } from 'enzyme'
import useRating from '../useRating'
import { act, renderHook } from '@testing-library/react-hooks'

describe('Rating', () => {
  it('rating hook', () => {
    const initValue = 3
    let valueRating = initValue
    const newRate = 4
    const handleRatingFilter = jest
      .fn()
      .mockImplementation(value => (valueRating = value))

    const { result } = renderHook(() =>
      useRating(initValue, handleRatingFilter)
    )

    expect(result.current[0]).toEqual(initValue)
    act(() => {
      result.current[1](null, newRate)
    })

    expect(handleRatingFilter).toHaveBeenCalled()
    expect(valueRating).toEqual(newRate)

    act(() => {
      result.current[1]()
    })

    expect(valueRating).toEqual(newRate)
  })

  it('rating', () => {
    const appRating = mount(<Rating value={4.3} name="test rating" />)

    expect(appRating).toMatchSnapshot()
  })
})
