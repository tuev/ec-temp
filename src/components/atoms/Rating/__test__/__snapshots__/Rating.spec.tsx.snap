// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating rating 1`] = `
<Rating
  name="test rating"
  value={4.3}
>
  <Styled(WithStyles(ForwardRef(Rating)))
    name="app-rating"
    onChange={[Function]}
    value={4.3}
  >
    <WithStyles(ForwardRef(Rating))
      className=""
      name="app-rating"
      onChange={[Function]}
      value={4.3}
    >
      <ForwardRef(Rating)
        className=""
        classes={
          Object {
            "decimal": "MuiRating-decimal",
            "disabled": "Mui-disabled",
            "focusVisible": "Mui-focusVisible",
            "icon": "MuiRating-icon",
            "iconActive": "MuiRating-iconActive",
            "iconEmpty": "MuiRating-iconEmpty",
            "iconFilled": "MuiRating-iconFilled",
            "iconFocus": "MuiRating-iconFocus",
            "iconHover": "MuiRating-iconHover",
            "label": "MuiRating-label",
            "pristine": "MuiRating-pristine",
            "readOnly": "MuiRating-readOnly",
            "root": "MuiRating-root",
            "sizeLarge": "MuiRating-sizeLarge",
            "sizeSmall": "MuiRating-sizeSmall",
            "visuallyhidden": "MuiRating-visuallyhidden",
          }
        }
        name="app-rating"
        onChange={[Function]}
        value={4.3}
      >
        <span
          aria-label={null}
          className="MuiRating-root "
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          role={null}
        >
          <label
            className="MuiRating-label"
            htmlFor="app-rating-1"
          >
            <IconContainer
              className="MuiRating-icon MuiRating-iconFilled"
              value={1}
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
              >
                <ForwardRef
                  fontSize="inherit"
                >
                  <WithStyles(ForwardRef(SvgIcon))
                    fontSize="inherit"
                  >
                    <ForwardRef(SvgIcon)
                      classes={
                        Object {
                          "colorAction": "MuiSvgIcon-colorAction",
                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                          "colorError": "MuiSvgIcon-colorError",
                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                          "root": "MuiSvgIcon-root",
                        }
                      }
                      fontSize="inherit"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </ForwardRef(SvgIcon)>
                  </WithStyles(ForwardRef(SvgIcon))>
                </ForwardRef>
              </span>
            </IconContainer>
            <span
              className="MuiRating-visuallyhidden"
            >
              1 Star
            </span>
          </label>
          <input
            checked={false}
            className="MuiRating-visuallyhidden"
            id="app-rating-1"
            name="app-rating"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value={1}
          />
          <label
            className="MuiRating-label"
            htmlFor="app-rating-2"
          >
            <IconContainer
              className="MuiRating-icon MuiRating-iconFilled"
              value={2}
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
              >
                <ForwardRef
                  fontSize="inherit"
                >
                  <WithStyles(ForwardRef(SvgIcon))
                    fontSize="inherit"
                  >
                    <ForwardRef(SvgIcon)
                      classes={
                        Object {
                          "colorAction": "MuiSvgIcon-colorAction",
                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                          "colorError": "MuiSvgIcon-colorError",
                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                          "root": "MuiSvgIcon-root",
                        }
                      }
                      fontSize="inherit"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </ForwardRef(SvgIcon)>
                  </WithStyles(ForwardRef(SvgIcon))>
                </ForwardRef>
              </span>
            </IconContainer>
            <span
              className="MuiRating-visuallyhidden"
            >
              2 Stars
            </span>
          </label>
          <input
            checked={false}
            className="MuiRating-visuallyhidden"
            id="app-rating-2"
            name="app-rating"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value={2}
          />
          <label
            className="MuiRating-label"
            htmlFor="app-rating-3"
          >
            <IconContainer
              className="MuiRating-icon MuiRating-iconFilled"
              value={3}
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
              >
                <ForwardRef
                  fontSize="inherit"
                >
                  <WithStyles(ForwardRef(SvgIcon))
                    fontSize="inherit"
                  >
                    <ForwardRef(SvgIcon)
                      classes={
                        Object {
                          "colorAction": "MuiSvgIcon-colorAction",
                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                          "colorError": "MuiSvgIcon-colorError",
                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                          "root": "MuiSvgIcon-root",
                        }
                      }
                      fontSize="inherit"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </ForwardRef(SvgIcon)>
                  </WithStyles(ForwardRef(SvgIcon))>
                </ForwardRef>
              </span>
            </IconContainer>
            <span
              className="MuiRating-visuallyhidden"
            >
              3 Stars
            </span>
          </label>
          <input
            checked={false}
            className="MuiRating-visuallyhidden"
            id="app-rating-3"
            name="app-rating"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value={3}
          />
          <label
            className="MuiRating-label"
            htmlFor="app-rating-4"
          >
            <IconContainer
              className="MuiRating-icon MuiRating-iconFilled"
              value={4}
            >
              <span
                className="MuiRating-icon MuiRating-iconFilled"
              >
                <ForwardRef
                  fontSize="inherit"
                >
                  <WithStyles(ForwardRef(SvgIcon))
                    fontSize="inherit"
                  >
                    <ForwardRef(SvgIcon)
                      classes={
                        Object {
                          "colorAction": "MuiSvgIcon-colorAction",
                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                          "colorError": "MuiSvgIcon-colorError",
                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                          "root": "MuiSvgIcon-root",
                        }
                      }
                      fontSize="inherit"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </ForwardRef(SvgIcon)>
                  </WithStyles(ForwardRef(SvgIcon))>
                </ForwardRef>
              </span>
            </IconContainer>
            <span
              className="MuiRating-visuallyhidden"
            >
              4 Stars
            </span>
          </label>
          <input
            checked={true}
            className="MuiRating-visuallyhidden"
            id="app-rating-4"
            name="app-rating"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value={4}
          />
          <label
            className="MuiRating-label"
            htmlFor="app-rating-5"
          >
            <IconContainer
              className="MuiRating-icon MuiRating-iconEmpty"
              value={5}
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <ForwardRef
                  fontSize="inherit"
                >
                  <WithStyles(ForwardRef(SvgIcon))
                    fontSize="inherit"
                  >
                    <ForwardRef(SvgIcon)
                      classes={
                        Object {
                          "colorAction": "MuiSvgIcon-colorAction",
                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                          "colorError": "MuiSvgIcon-colorError",
                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                          "root": "MuiSvgIcon-root",
                        }
                      }
                      fontSize="inherit"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                        />
                      </svg>
                    </ForwardRef(SvgIcon)>
                  </WithStyles(ForwardRef(SvgIcon))>
                </ForwardRef>
              </span>
            </IconContainer>
            <span
              className="MuiRating-visuallyhidden"
            >
              5 Stars
            </span>
          </label>
          <input
            checked={false}
            className="MuiRating-visuallyhidden"
            id="app-rating-5"
            name="app-rating"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value={5}
          />
        </span>
      </ForwardRef(Rating)>
    </WithStyles(ForwardRef(Rating))>
  </Styled(WithStyles(ForwardRef(Rating)))>
</Rating>
`;
